ReferralPlease/ReferralPlease/ViewController.swift

```diff
@@ -141,93 +158,136 @@ extension ViewController: WKNavigationDelegate {
         // Set the POST parameters.
         let postParams = "grant_type=" + grantType + "&code=" + authCode + "&redirect_uri=" + LinkedInConstants.REDIRECT_URI + "&client_id=" + LinkedInConstants.CLIENT_ID + "&client_secret=" + LinkedInConstants.CLIENT_SECRET
         let postData = postParams.data(using: String.Encoding.utf8)
-        let request = NSMutableURLRequest(url: URL(string: LinkedInConstants.TOKENURL)!)
-        request.httpMethod = "POST"
-        request.httpBody = postData
-        request.addValue("application/x-www-form-urlencoded;", forHTTPHeaderField: "Content-Type")
-        let session = URLSession(configuration: URLSessionConfiguration.default)
-        let task: URLSessionDataTask = session.dataTask(with: request as URLRequest) { (data, response, error) -> Void in
-            let statusCode = (response as! HTTPURLResponse).statusCode
-            if statusCode == 200 {
-                let results = try! JSONSerialization.jsonObject(with: data!, options: .allowFragments) as? [AnyHashable: Any]
-
-                let accessToken = results?["access_token"] as! String
-                print("accessToken is: \(accessToken)")
-
-                let expiresIn = results?["expires_in"] as! Int
-                print("expires in: \(expiresIn)")
-
-                // Get user's id, first name, last name, profile pic url
-                self.fetchLinkedInUserProfile(accessToken: accessToken)
+        if let url = URL(string: LinkedInConstants.TOKENURL) {
+            let request = NSMutableURLRequest(url: url)
+            request.httpMethod = "POST"
+            request.httpBody = postData
+            request.addValue("application/x-www-form-urlencoded;", forHTTPHeaderField: "Content-Type")
+            let session = URLSession(configuration: URLSessionConfiguration.default)
+            let task: URLSessionDataTask = session.dataTask(with: request as URLRequest) { (data, response, error) -> Void in
+                do {
+                let statusCode = (response as? HTTPURLResponse)?.statusCode
+                if statusCode == 200 {
+                    if let jsonData = data{
+                    let results = try JSONSerialization.jsonObject(with: jsonData, options: .allowFragments) as? [AnyHashable: Any]
+
+                    let accessToken = results?["access_token"] as? String
+                        print("accessToken is: \(accessToken ?? "")")
+
+                    let expiresIn = results?["expires_in"] as? Int
+                        print("expires in: \(expiresIn ?? nil)")
+
+                    // Get user's id, first name, last name, profile pic url
+                        if let access = accessToken{
+                    self.fetchLinkedInUserProfile(accessToken: access)
```

Comment:
I'm quite annoyed by the indentation of this block. Is it possible that you can fix that?